subsys/mgmt/osdp/src/osdp_cp.c

```diff
@@ -636,34 +633,35 @@ static int cp_phy_state_update(struct osdp_pd *pd)
 		/* fall-thru */
 	case OSDP_CP_PHY_STATE_SEND_CMD:
 		if ((cp_send_command(pd)) < 0) {
-			LOG_ERR(TAG "send command error");
+			LOG_ERR("Failed to send CMD(%d)", pd->cmd_id);
 			pd->phy_state = OSDP_CP_PHY_STATE_ERR;
 			ret = OSDP_CP_ERR_GENERIC;
 			break;
 		}
+		ret = OSDP_CP_ERR_INPROG;
 		pd->phy_state = OSDP_CP_PHY_STATE_REPLY_WAIT;
 		pd->rx_buf_len = 0; /* reset buf_len for next use */
 		pd->phy_tstamp = osdp_millis_now();
 		break;
 	case OSDP_CP_PHY_STATE_REPLY_WAIT:
-		tmp = cp_process_reply(pd);
-		if (tmp == 0) { /* success */
+		rc = cp_process_reply(pd);
+		if (rc == OSDP_CP_ERR_NONE) {
 			pd->phy_state = OSDP_CP_PHY_STATE_CLEANUP;
 			break;
 		}
-		if (tmp == OSDP_CP_ERR_RETRY_CMD) {
-			LOG_INF(TAG "PD busy; retry last command");
+		if (rc == OSDP_CP_ERR_RETRY_CMD) {
+			LOG_INF("PD busy; retry last command");
 			pd->phy_tstamp = osdp_millis_now();
 			pd->phy_state = OSDP_CP_PHY_STATE_WAIT;
-			ret = 2;
```

Comment:
No, return value of 2 was used to indicate a safe point for caller to yield channel to another PD since we are in between commands. Now it is called `OSDP_CP_ERR_CAN_YIELD` and is returned by initialising ret at function entry.